front/src/common/Map/Layers/InfraObjectLayers/InfraObjectLayers.tsx

```diff
@@ -18,84 +23,104 @@ type InfraObjectLayersProps = {
   infraId: number;
   mapStyle: MapStyle;
   hoveredOperationalPointId?: string;
+  compact: boolean;
 };
 
 const InfraObjectLayers = ({
   infraId,
   mapStyle,
   hoveredOperationalPointId,
-}: InfraObjectLayersProps) => (
-  <>
-    <TracksGeographic
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.TRACKS.GROUP]}
-      infraID={infraId}
-    />
+  compact,
+}: InfraObjectLayersProps) => {
+  const layersSettings = useSelector(compact ? getStdcmSelectedLayers : getLayersSettings);
 
-    <Routes
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.ROUTES.GROUP]}
-      infraID={infraId}
-    />
+  return (
+    <>
+      <TracksGeographic
+        colors={colors[mapStyle]}
+        layerOrder={LAYER_GROUPS_ORDER[LAYERS.TRACKS.GROUP]}
+        infraID={infraId}
+      />
 
-    <OperationalPoints
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.OPERATIONAL_POINTS.GROUP]}
-      operationnalPointId={hoveredOperationalPointId}
-      infraID={infraId}
-    />
+      <Routes
+        colors={colors[mapStyle]}
+        layerOrder={LAYER_GROUPS_ORDER[LAYERS.ROUTES.GROUP]}
+        infraID={infraId}
+      />
 
-    <Electrifications
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.ELECTRIFICATIONS.GROUP]}
-      infraID={infraId}
-    />
+      <OperationalPoints
+        colors={colors[mapStyle]}
+        layerOrder={LAYER_GROUPS_ORDER[LAYERS.OPERATIONAL_POINTS.GROUP]}
+        operationnalPointId={hoveredOperationalPointId}
+        infraID={infraId}
+        layersSettings={layersSettings}
+        overrideStore
+      />
 
-    <NeutralSections
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.DEAD_SECTIONS.GROUP]}
-      infraID={infraId}
-    />
+      <Electrifications
+        colors={colors[mapStyle]}
+        layerOrder={LAYER_GROUPS_ORDER[LAYERS.ELECTRIFICATIONS.GROUP]}
+        infraID={infraId}
+        layersSettings={layersSettings}
+      />
```

Comment:
You have modified a lot of layers to add the layerSettings, but actually it is only used to know if the layer is enabled or not.
You don't need to pass the layersSettings to the layer actually, you can simply display the layer if it is enabled like this:
```suggestion
       {layersSettings.electrifications && (
      <Electrifications
        colors={colors[mapStyle]}
        layerOrder={LAYER_GROUPS_ORDER[LAYERS.ELECTRIFICATIONS.GROUP]}
        infraID={infraId}
      />
    )}
```

You can do this for all the layers ;)